tools/src/test/java/org/apache/kafka/trogdor/workload/ExternalCommandWorkerTest.java

```diff
@@ -0,0 +1,196 @@
+        if (OperatingSystem.IS_WINDOWS) return;
+        ExternalCommandWorker worker =
+            new ExternalCommandWorkerBuilder("trueTask").command("true").build();
+        KafkaFutureImpl<String> doneFuture = new KafkaFutureImpl<>();
+        worker.start(null, new AgentWorkerStatusTracker(), doneFuture);
+        assertEquals("", doneFuture.get());
+        worker.stop(null);
+    }
+
+    /**
+     * Test running a process which exits unsuccessfully-- in this case, /bin/false.
+     */
+    @Test
+    public void testProcessWithFailedExit() throws Exception {
+        if (OperatingSystem.IS_WINDOWS) return;
+        ExternalCommandWorker worker =
+            new ExternalCommandWorkerBuilder("falseTask").command("false").build();
+        KafkaFutureImpl<String> doneFuture = new KafkaFutureImpl<>();
+        worker.start(null, new AgentWorkerStatusTracker(), doneFuture);
+        assertEquals("exited with return code 1", doneFuture.get());
+        worker.stop(null);
+    }
+
+    /**
+     * Test attempting to run an exeutable which doesn't exist.
+     * We use a path which starts with /dev/null, since that should never be a
+     * directory in UNIX.
+     */
+    @Test
+    public void testProcessNotFound() throws Exception {
+        ExternalCommandWorker worker =
+            new ExternalCommandWorkerBuilder("notFoundTask").
+                command("/dev/null/non/existent/script/path").build();
+        KafkaFutureImpl<String> doneFuture = new KafkaFutureImpl<>();
+        worker.start(null, new AgentWorkerStatusTracker(), doneFuture);
+        String errorString = doneFuture.get();
+        assertTrue(errorString.startsWith("Unable to start process"));
+        worker.stop(null);
+    }
+
+    /**
+     * Test running a process which times out.  We will send it a SIGTERM.
+     */
+    @Test
+    public void testProcessStop() throws Exception {
+        if (OperatingSystem.IS_WINDOWS) return;
+        ExternalCommandWorker worker =
+            new ExternalCommandWorkerBuilder("testStopTask").
+                command("sleep", "3600000").build();
+        KafkaFutureImpl<String> doneFuture = new KafkaFutureImpl<>();
+        worker.start(null, new AgentWorkerStatusTracker(), doneFuture);
+        worker.stop(null);
+        // We don't check the numeric return code, since that will vary based on
+        // platform.
+        assertTrue(doneFuture.get().startsWith("exited with return code "));
+    }
+
+    /**
+     * Test running a process which needs to be force-killed.
+     */
+    @Test
+    public void testProcessForceKillTimeout() throws Exception {
+        if (OperatingSystem.IS_WINDOWS) return;
+        File tempFile = null;
+        try {
+            tempFile = TestUtils.tempFile();
+            try (OutputStream stream = Files.newOutputStream(tempFile.toPath())) {
+                for (String line : new String[] {
+                    "echo hello world\n",
```

Comment:
Maybe move this to a test resource? Escaped strings are a pain to update and easy to screw up

Reply:
Java is adding literal strings soon that won't require escaping here.  probably easier to leave as-is?  What do you think?

Reply:
It's not a big deal either way, so let's leave it as-is 👍